packages/scaffold/src/views/w3m-connecting-siwe-view/index.ts

```diff
@@ -0,0 +1,110 @@
+import { CoreHelperUtil, OptionsController, RouterController, StorageUtil } from '@web3modal/core'
+import { customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import { state } from 'lit/decorators.js'
+
+@customElement('w3m-connecting-siwe-view')
+export class W3mConnectingSiweView extends LitElement {
+  // -- Members -------------------------------------------- //
+  private usubscribe: (() => void)[] = []
+
+  // -- State & Properties --------------------------------- //
+  @state() private dappIcons = OptionsController.state.metadata?.icons
+
+  @state() private dappUrl = OptionsController.state.metadata?.url
+
+  @state() private dappName = OptionsController.state.metadata?.name
+
+  public constructor() {
+    super()
+
+    this.usubscribe.push(
+      OptionsController.subscribeKey('metadata', metadata => {
+        if (metadata?.icons) {
+          this.dappIcons = metadata.icons
+        }
+        if (metadata?.url) {
+          this.dappUrl = metadata.url
+        }
+        if (metadata?.name) {
+          this.dappName = metadata.name
+        }
+      })
+    )
+  }
+
+  // -- Render -------------------------------------------- //
+
+  public override render() {
+    const imageUrl = StorageUtil.getImageUrl()
```

Comment:
This can can also be a readonly global, instead of accessing localStorage within each render call.